### ISSUE TITLE
generation: 0 prediction for all test values 

### ISSUE THREAD

**Issue by GaganaB** [EpistasisLab/tpot#1043]
I tried the following configuration:
 tpot = TPOTClassifier(verbosity=2, max_time_mins=2, max_eval_time_mins=0.04, population_size=40)
on binary classification problem using: https://github.com/EpistasisLab/tpot/blob/master/tutorials/Titanic_Kaggle.ipynb
But tpot generates 0 prediction for all test values.  

My training set has 86 rows and 412 columns. 
My test set however is inflated to 119748 rows and 411 columns.

**Comment by weixuanfu** [comment]
Could you please provide more details about this issue like the codes that you used or error messages? 
My first guess is that `max_eval_time_mins=0.04` may be too small for your dataset. Please reset it to default value 5 and then have a try.

Also, why the test set has 1 less column than the training set? Is the missing column a target column?

**Comment by GaganaB** [comment]
https://colab.research.google.com/drive/1SapPBGJu-Woa9PE39ug1ckhWWUF0PLZ3
Please find my colab notebook here. 
I tried increasing the max_eval_time_mins but still the same result. 
And Yes, the missing column is the target column, so, I have merged it with the original prediction file for tpot. I think that could be the issue but I'm not sure what to do.

**Comment by weixuanfu** [comment]
I checked the notebook, the training process is not right. Like the code box below:

```python
tpot = TPOTClassifier(verbosity=2, max_time_mins=2, max_eval_time_mins=0.04, population_size=40)
t = train["Class"].values
tpot.fit(train, t)
```
It means that the target column "Class" was in the `train` dataframe, which is not right to put dependent target column in `X`. You should change it to

```python
tpot = TPOTClassifier(verbosity=2, max_time_mins=2, population_size=40)
t = train["Class"].values
X = train.drop("Class", axis=1).values
tpot.fit(X, t)
```

Also for using exported codes, you set all the class code to 0 in the box below. And then fit the model via `exported_pipeline.fit(test, test["Class"].values)` so that is the reason why the predictions of test set are all 0.

```python
test = pd.read_csv("test.csv")
test['Class'] = [0]*119748
```

The correct way is to refit the exported model on train set (note: please also drop "Class" before the call `exported_pipeline.fit(X, y)`) and then predict it on test set.

**Comment by GaganaB** [comment]
@weixuanfu  This works perfectly. Thank you very much. Its giving me a performance auc of around 0.74. Do you have any recommendations on how I can do better please? Thank you.
